pkg/tiflowapi/client.go

```diff
@@ -23,21 +22,32 @@ func GetMasterClient(cli client.Client, namespace, tcName, podName string, tlsEn
 		tlsConfig, err = GetTLSConfig(cli, namespace, util.ClusterClientTLSSecretName(tcName))
 		if err != nil {
 			klog.Errorf("Unable to get tls config for tiflow cluster %q, master client may not work: %v", tcName, err)
-			return NewMasterClient(MasterClientURL(namespace, tcName, podName, scheme), DefaultTimeout, tlsConfig)
+			return NewMasterClient(MasterClientURL(namespace, tcName, podName, scheme, hetero), DefaultTimeout, tlsConfig)
 		}
 
-		return NewMasterClient(MasterClientURL(namespace, tcName, podName, scheme), DefaultTimeout, tlsConfig)
+		return NewMasterClient(MasterClientURL(namespace, tcName, podName, scheme, hetero), DefaultTimeout, tlsConfig)
 	}
 
-	return NewMasterClient(MasterClientURL(namespace, tcName, podName, scheme), DefaultTimeout, nil)
+	return NewMasterClient(MasterClientURL(namespace, tcName, podName, scheme, hetero), DefaultTimeout, nil)
+}
+
+type Heterogeneous struct {
```

Comment:
could you leave a comment here to show why we need this struct?

Reply:
Normally, we can get the Executors information directly from the basic master.
However, in the heterogeneous case, we need to change the masterHost because the current executor cluster and the basic master cluster are not in the same namespace